### ISSUE TITLE
Fix 'ggl::gZero' in 'ggl::matrix' Compiler Errors

### ISSUE THREAD

**Issue by churay** [churay/ggl#20]
Currently, attempting to use the recently added `ggl::gZero` template variable in the source code for the `ggl::matrix` class results in the following error when compiling test cases:

```
$ clang++ -std=c++1y -stdlib=libc++ -Wall -Werror -g -O0 -I. -I./lib test/matrix.cpp -c -o bin/matrix.to
In file included from test/matrix.cpp:2:
In file included from ./src/matrix.hpp:10:
./src/consts.hpp:10:36: error: variable 'ggl::gZero<int>' has internal linkage but is not defined [-Werror,-Wundefined-internal]
    template <class T> constexpr T gZero{ static_cast<T>(0) };
                                   ^
./src/matrix.cpp:29:30: note: used here
        (*this)[eIdx] = ggl::gZero<T>;
                             ^
1 error generated.
```

This should be fixed so that `ggl::gZero` can be used in the `ggl::matrix` class to replace the static zero and one values that are currently being used.

**Comment by churay** [comment]
The information in [this thread about template variables](https://groups.google.com/a/isocpp.org/forum/#!topic/std-proposals/ZTw3W3vjWWs) may help in resolving this issue.

**Comment by churay** [comment]
Fixed with the changes in commit 90e1e8499ba5508e1b9878fb12677577f0f70a71.
